File: PolicyPlus/EditPolDelete.vb
Public Class EditPolDelete
    Public Function PresentDialog(ContainerKey As String) As DialogResult
        OptClearFirst.Checked = False
        OptDeleteOne.Checked = False
        OptPurge.Checked = False
        TextKey.Text = ContainerKey
        TextValueName.Text = ""
        Return ShowDialog()
    End Function
    Private Sub ButtonOK_Click(sender As Object, e As EventArgs) Handles ButtonOK.Click
        If OptClearFirst.Checked Or OptPurge.Checked Then DialogResult = DialogResult.OK
        If OptDeleteOne.Checked Then
            If TextValueName.Text = "" Then
                MsgBox("You must enter a value name.", MsgBoxStyle.Exclamation)
                Exit Sub
            End If
            DialogResult = DialogResult.OK
        End If
    End Sub
    Private Sub EditPolDelete_KeyDown(sender As Object, e As KeyEventArgs) Handles Me.KeyDown
        If e.KeyCode = Keys.Escape Then DialogResult = DialogResult.Cancel
    End Sub
    Sub ChoiceChanged(sender As Object, e As EventArgs) Handles OptClearFirst.CheckedChanged, OptDeleteOne.CheckedChanged, OptPurge.CheckedChanged
        TextValueName.Enabled = OptDeleteOne.Checked
    End Sub
End Class

File: PolicyPlus/EditPolDelete.cs
using System;
using System.Windows.Forms;
using Microsoft.VisualBasic;

namespace PolicyPlus
{
    public partial class EditPolDelete
    {
        public EditPolDelete()
        {
            InitializeComponent();
        }
        public DialogResult PresentDialog(string ContainerKey)
        {
            OptClearFirst.Checked = false;
            OptDeleteOne.Checked = false;
            OptPurge.Checked = false;
            TextKey.Text = ContainerKey;
            TextValueName.Text = "";
            return ShowDialog();
        }
        private void ButtonOK_Click(object sender, EventArgs e)
        {
            if (OptClearFirst.Checked | OptPurge.Checked)
                DialogResult = DialogResult.OK;
            if (OptDeleteOne.Checked)
            {
                if (string.IsNullOrEmpty(TextValueName.Text))
                {
                    Interaction.MsgBox("You must enter a value name.", MsgBoxStyle.Exclamation);
                    return;
                }
                DialogResult = DialogResult.OK;
            }
        }
        private void EditPolDelete_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.Escape)
                DialogResult = DialogResult.Cancel;
        }
        public void ChoiceChanged(object sender, EventArgs e)
        {
            TextValueName.Enabled = OptDeleteOne.Checked;
        }
    }
}